MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -435,6 +453,225 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputCloseSurfaceModel(vtkMRMLMarkupsT
+  double variationMediumOBBAxis[ 3 ]  = { 0.0, 0.0, 0.0 };
+  double variationMinimumOBBAxis[ 3 ] = { 0.0, 0.0, 0.0 };
+  double relativeAxisSizes[ 3 ] = { 0.0, 0.0, 0.0 }; // unused, the values represented herein are unclear
+  vtkSmartPointer<vtkOBBTree> obbTree = vtkSmartPointer<vtkOBBTree>::New();
+  obbTree->ComputeOBB( points, cornerOBBOrigin, variationMaximumOBBAxis, variationMediumOBBAxis, variationMinimumOBBAxis, relativeAxisSizes );
+  
+  // now to store the desired results in the appropriate output variables
+  // must check each axis to make sure it was actually computed (non-zero)
+  // do the major axis
+  if ( vtkMath::Norm( variationMaximumOBBAxis ) < COMPARE_TO_ZERO_TOLERANCE )
+  {
+    // there is no variation in the points whatsoever.
+    // i.e. all points are in a single position.
+    // return arbitrary orthonormal axes (the standard axes will do).
+    outputPlaneMajorAxis[ 0 ] = 1.0;
+    outputPlaneMajorAxis[ 1 ] = 0.0;
+    outputPlaneMajorAxis[ 2 ] = 0.0;
+    outputPlaneMinorAxis[ 0 ] = 0.0;
+    outputPlaneMinorAxis[ 1 ] = 1.0;
+    outputPlaneMinorAxis[ 2 ] = 0.0;
+    outputPlaneNormal[ 0 ] = 0.0;
+    outputPlaneNormal[ 1 ] = 0.0;
+    outputPlaneNormal[ 2 ] = 1.0;
+    return;
+  }
+
+  outputPlaneMajorAxis[ 0 ] = variationMaximumOBBAxis[ 0 ];
+  outputPlaneMajorAxis[ 1 ] = variationMaximumOBBAxis[ 1 ];
+  outputPlaneMajorAxis[ 2 ] = variationMaximumOBBAxis[ 2 ];
+  vtkMath::Normalize( outputPlaneMajorAxis );
+
+  // do the minor axis
+  if ( vtkMath::Norm( variationMediumOBBAxis ) < COMPARE_TO_ZERO_TOLERANCE )
+  {
+    // the points are colinear.
+    // there is no distinct normal or minor axis relative to the plane.
+    // any two perpendicular orthonormal vectors will do for the remaining axes.
+    double thetaAngle = 0.0; // this can be arbitrary
+    vtkMath::Perpendiculars( outputPlaneMajorAxis, outputPlaneMinorAxis, outputPlaneNormal, thetaAngle );
+    return;
+  }
+  outputPlaneMinorAxis[ 0 ] = variationMediumOBBAxis[ 0 ];
+  outputPlaneMinorAxis[ 1 ] = variationMediumOBBAxis[ 1 ];
+  outputPlaneMinorAxis[ 2 ] = variationMediumOBBAxis[ 2 ];
+  vtkMath::Normalize( outputPlaneMinorAxis );
+
+  // do the plane normal
+  if ( vtkMath::Norm( variationMinimumOBBAxis ) < COMPARE_TO_ZERO_TOLERANCE )
+  {
+    // all points lie exactly on a plane.
+    // the remaining perpendicular vector found using cross product will be fine.
+    vtkMath::Cross( outputPlaneMajorAxis, outputPlaneMinorAxis, outputPlaneNormal );
+    return;
+  }
+  outputPlaneNormal[ 0 ] = variationMinimumOBBAxis[ 0 ];
+  outputPlaneNormal[ 1 ] = variationMinimumOBBAxis[ 1 ];
+  outputPlaneNormal[ 2 ] = variationMinimumOBBAxis[ 2 ];
+  vtkMath::Normalize( outputPlaneNormal );
+}
+
+//------------------------------------------------------------------------------
+vtkSlicerMarkupsToModelLogic::PointArrangementEnum vtkSlicerMarkupsToModelLogic::ComputePointArrangement( vtkPoints* points, double planeNormal[ 3 ], double lineAxis[ 3 ] )
```

Comment:
Extrusion amount was determined through testing. I think I wrote something to that effect in a comment, but perhaps I need to clarify:

The problem is if the extrusion amount is reduced below about 1/10 of the maximum extend, then the delaunay filter can fail and not produce any output. So you end up with weird "gray zones" between planar and non-planar surfaces where you get no output at all.